src/rtr.rs

```diff
@@ -147,6 +150,32 @@ impl RtrStream {
             metrics
         })
     }
+
+    #[cfg(unix)]
+    fn set_keepalive(
+        sock: &TcpStream, duration: Duration
+    ) -> Result<(), io::Error>{
+        use std::convert::TryFrom;
+        use std::os::unix::io::AsRawFd;
+        use nix::sys::socket::{setsockopt, sockopt};
+
+        (|fd, duration: Duration| {
+            setsockopt(
+                fd, sockopt::TcpKeepInterval,
+                &u32::try_from(duration.as_secs()).unwrap_or(u32::MAX)
```

Comment:
Wouldn't it make more sense to use a realistic/common value instead of `u32::MAX` to fall back on? Because with an interval time that large, I suspect that the keepalive will not work in practice even though it is technically enabled.

fwiw:
```
$ cat /proc/sys/net/ipv4/tcp_keepalive_time
7200
```